MoltenVK/MoltenVK/Commands/MVKCommandPipelineStateFactoryShaderSource.h

```diff
@@ -402,5 +402,20 @@ kernel void accumulateOcclusionQueryResults(device VisibilityBuffer& dest [[buff
     if (dest.count < max(oldDestCount, src.count)) { dest.countHigh++; }                                        \n\
 }                                                                                                               \n\
                                                                                                                 \n\
+kernel void remapVertices(device const uint* vertexIndices [[ buffer(0) ]],                                     \n\
+                          device uint* remappedVertices [[ buffer(1) ]],                                        \n\
+                          uint numVertices [[ thread_position_in_grid ]]) {                                     \n\
```

Comment:
I understand this is a WIP, but I'd like to understand how this shader works.

`thread_position_in_grid` will be a unique index for each instance, and I assume, one for each vertex that is to be remapped? 

I'm not sure you want to be looping here, as the higher instances will tromp on the lower instances. If `thread_position_in_grid` does represent the original index, then the calculation of `index` could just be done once per shader instance.

Reply:
To be entirely honest once I started working on this PR I quickly realised I was out of my depth. 😅 I may just close this PR as I doubt I can finish it without hours of digging and probably countless questions.

Reply:
@IsaacMarovitz can the code from rpcs3 - https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/Common/BufferUtils.cpp#L464 - or xenia - https://github.com/xenia-project/xenia/blob/master/src/xenia/gpu/primitive_processor.cc - help you?

Reply:
> I may just close this PR

Leave it open as WIP. You have some great ideas here, and it's a good start! You can return to it later, or I or someone else could pick it up from here. So thanks very much for your contribution, and for at the very least kicking this off!

With that in mind, if anyone is going to work on this, just post here that you are, so that we can all coordinate, and not duplicate efforts.

Reply:
> @IsaacMarovitz can the code from rpcs3 - https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/Common/BufferUtils.cpp#L464 - or xenia - https://github.com/xenia-project/xenia/blob/master/src/xenia/gpu/primitive_processor.cc - help you?

Thanks for the resources I'll take a look through and see if I can gleam anything useful 😄